ros_nodes/yumi_arm_service.py

```diff
@@ -29,19 +34,41 @@ def handle_request(req):
         func = pickle.loads(req.func)
         args = pickle.loads(req.args)
         kwargs = pickle.loads(req.kwargs)
-        
+
         if func == '__getattribute__':
             if verbose:
-                rospy.loginfo("Handling request to get attribute {0} for {1} arm".format(args, name))
+                rospy.loginfo(
+                    "Handling request to get attribute {0} for {1} arm".format(
+                        args, name))
             return ROSYumiArmResponse(pickle.dumps(getattr(arm, args)))
         else:
             if verbose:
-                rospy.loginfo("Handling request to call method {0} for {1} arm".format(func, name))
-            return ROSYumiArmResponse(pickle.dumps(yumi_methods[func](arm, *args, **kwargs)))
-    
+                rospy.loginfo(
+                    "Handling request to call method {0} for {1} arm".format(
+                        func, name))
+            return ROSYumiArmResponse(
+                pickle.dumps(yumi_methods[func](arm, *args, **kwargs)))
+
     s = rospy.Service('{0}_arm'.format(name), ROSYumiArm, handle_request)
     rospy.loginfo("{0} arm is ready".format(name))
 
+    joint_state_pub = rospy.Publisher(
+        '/joint_states', JointState, queue_size=10)
```

Comment:
This ignores namespace, recommend making it relative instead (`'joint_states'` instead of `'/joint_states'`)

Also recommend separate topics for left and right arms (and removing the `name[0]` postfix on line 65), but this is personal preference.